Add doc to update configuration after installation (#296)
harvester/docs: i18n/zh/docusaurus-plugin-content-docs/version-v1.1/install/update-harvester-configuration.md

---
sidebar_position: 6
sidebar_label: Update Harvester Configuration
title: "Update Harvester Configuration After Installation"
keywords:
  - Harvester configuration
  - Configuration
Description: How to update Harvester configuration after installation
---

> Harvester's OS has an immutable design, which means most files in the  OS revert to their pre-configured state after a reboot. Harvester OS loads the pre-configured values of system components from configuration files during the boot time. To update a configuration, the user must first update the runtime value in the system and update configuration files to persist changes between reboots. This doc describes how to edit some most-requested Harvester configurations.

Reviewer: ```suggestion
Harvester's OS has an immutable design, which means most files in the  OS revert to their pre-configured state after a reboot. The Harvester OS loads the pre-configured values of system components from configuration files during the boot time. 

This page describes how to edit some of the most-requested Harvester configurations. To update a configuration, you must first update the runtime value in the system and then update configuration files to make the changes persistent between reboots. 
```

:::note

> If you upgrade from a version before `1.1.2`, the cloud-init file in examples will be `/oem/99_custom.yaml`. Please substitute the value if needed.

Reviewer: ```suggestion
If you upgrade from a version before `v1.1.2`, the `cloud-init` file in examples will be `/oem/99_custom.yaml`. Please substitute the value if needed.
```

:::

## DNS servers

### Runtime change

> 1. Log in to a Harvester node and become root. See this [page](../troubleshooting/os.md#how-to-log-into-a-harvester-node).

Reviewer: ```suggestion
1. Log in to a Harvester node and become root. See [how to log into a Harvester node](../troubleshooting/os.md#how-to-log-into-a-harvester-node) for more details.
```

> 1. Edit `/etc/sysconfig/network/config` and update the line (use space if there are multiple servers):

Reviewer: ```suggestion
1. Edit `/etc/sysconfig/network/config` and update the following line. Use a space to separate DNS server addresses if there are multiple servers.
```

    ```
    NETCONFIG_DNS_STATIC_SERVERS="8.8.8.8 1.1.1.1"
    ```

> 1. Reload the configuration with the command:

Reviewer: ```suggestion
1. Update and reload the configuration with the following command:
```

    ```
    netconfig update
    ```

> 1. Check the file `/etc/resolv.conf` contains the correct DNS servers:

Reviewer: ```suggestion
1. Confirm the file `/etc/resolv.conf` contains the correct DNS servers with the `cat` command:
```

    ```
    cat /etc/resolv.conf
    ```

### Configuration persistence

> 1. Backup the elemental cloud-init file: `/oem/90_custom.yaml`:

Reviewer: ```suggestion
1. Backup the elemental `cloud-init` file `/oem/90_custom.yaml` as follows:
```

    ```
    cp /oem/90_custom.yaml /oem/install/90_custom.yaml.$(date --iso-8601=minutes)
    ```

> 1. Edit `/oem/90_custom.yaml` and update the value under yaml path `stages.initramfs[0].commands`. The `commands` array must contain a line to manipulate the `NETCONFIG_DNS_STATIC_SERVERS` config (Create the line if it doesn't exist). For example:

Reviewer: ```suggestion
1. Edit `/oem/90_custom.yaml` and update the value under the yaml path `stages.initramfs[0].commands`. The `commands` array must contain a line to manipulate the `NETCONFIG_DNS_STATIC_SERVERS` config. Add the line if the line doesn't exist. 

The following example adds a line to change the `NETCONFIG_DNS_STATIC_SERVERS` config:
```

    ```
    stages:
      initramfs:
        - commands:
            - sed -i 's/^NETCONFIG_DNS_STATIC_SERVERS.*/NETCONFIG_DNS_STATIC_SERVERS="8.8.8.8 1.1.1.1"/' /etc/sysconfig/network/config
    ```

>     Replace the name servers and save the file. Harvester will set up new servers after rebooting.

Reviewer: ```suggestion
    Replace the DNS server addresses and save the file. Harvester sets up new servers after rebooting.
```

## NTP servers

### Runtime change

> 1. Log in to a Harvester node and become root. See this [page](../troubleshooting/os.md#how-to-log-into-a-harvester-node).

Reviewer: ```suggestion
1. Log in to a Harvester node and become root. See [how to log into a Harvester node](../troubleshooting/os.md#how-to-log-into-a-harvester-node) for more details.
```

1. Edit `/etc/systemd/timesyncd.conf` and specify NTP servers in the `NTP=` setting:

    ```
    [Time]
    NTP = 0.suse.pool.ntp.org 1.suse.pool.ntp.org
    ```

1. Restart the `systemd-timesyncd.service` service:

    ```
    systemctl restart systemd-timesyncd.service
    ```

1. Display the timesync status:

    ```
    timedatectl timesync-status
    ```

### Configuration persistence

> 1. Backup the elemental cloud-init file: `/oem/90_custom.yaml`:

Reviewer: ```suggestion
1. Backup the elemental `cloud-init` file `/oem/90_custom.yaml` as follows:
```

    ```
    cp /oem/90_custom.yaml /oem/install/90_custom.yaml.$(date --iso-8601=minutes)
    ```

1. Edit `/oem/90_custom.yaml` and update the yaml path `stages.initramfs[0].timesyncd`. The `timesyncd` map must be in the following format:

    ```
    stages:
      initramfs:
      - ...
        timesyncd:
        NTP: 0.suse.pool.ntp.org 1.suse.pool.ntp.org
    ```

1. Edit `/oem/90_custom.yaml` and update the yaml path `stages.initramfs[0].systemctl.enable`. The array must have these two services presented: 

    ```
    stages:
      initramfs:
      - ...
        systemctl:
        enable:
            systemd-timesyncd
            systemd-time-wait-sync
        disable: []
        start: []
        mask: []
    ```

## SSH keys of user `rancher`

### Runtime change

> 1. Log in to a Harvester node as user `rancher`. See this [page](../troubleshooting/os.md#how-to-log-into-a-harvester-node).

Reviewer: ```suggestion
1. Log in to a Harvester node as user `rancher`. See [how to log into a Harvester node](../troubleshooting/os.md#how-to-log-into-a-harvester-node) for more details.
```

1. Edit `/home/rancher/.ssh/authorized_keys` to add or remove keys.

### Configuration persistence

> 1. Backup the elemental cloud-init file: `/oem/90_custom.yaml`:

Reviewer: ```suggestion
1. Backup the elemental `cloud-init` file `/oem/90_custom.yaml` as follows:
```

    ```
    cp /oem/90_custom.yaml /oem/install/90_custom.yaml.$(date --iso-8601=minutes)
    ```

1. Edit `/oem/90_custom.yaml` and update the yaml path `stages.network[0].authorized_keys.rancher`. Add or remove keys in the `rancher` array:

    ```
    stages:
      network:
      - ...
        authorized_keys:
          rancher:
          - key1
          - key2
    ```

## Password of user `rancher`

### Runtime change

> 1. Log in to a Harvester node as user `rancher`. See this [page](../troubleshooting/os.md#how-to-log-into-a-harvester-node).

Reviewer: ```suggestion
1. Log in to a Harvester node as user `rancher`. See [how to log into a Harvester node](../troubleshooting/os.md#how-to-log-into-a-harvester-node) for more details.
```

1. Run `passwd` to reset user `rancher`'s password.

### Configuration persistence

> 1. Backup the elemental cloud-init file: `/oem/90_custom.yaml`:

Reviewer: ```suggestion
1. Backup the elemental `cloud-init` file `/oem/90_custom.yaml` as follows:
```

    ```
    cp /oem/90_custom.yaml /oem/install/90_custom.yaml.$(date --iso-8601=minutes)
    ```

> 1. Edit `/oem/90_custom.yaml` and update the yaml path `stages.initramfs[0].users.rancher.passwd`. See the configuration [`os.password`](./harvester-configuration.md#ospassword) about how to specify the password in the encrypted form.

Reviewer: ```suggestion
1. Edit `/oem/90_custom.yaml` and update the yaml path `stages.initramfs[0].users.rancher.passwd`. Refer to the configuration [`os.password`](./harvester-configuration.md#ospassword) for details on how to specify the password in an encrypted form.
```

## Bonding slaves

You can update the slave interfaces of Harvester's management bonding interface `mgmt-bo`.

### Runtime change

> 1. Log in to a Harvester node and become root. See this [page](../troubleshooting/os.md#how-to-log-into-a-harvester-node).

Reviewer: ```suggestion
1. Log in to a Harvester node and become root. See [how to log into a Harvester node](../troubleshooting/os.md#how-to-log-into-a-harvester-node) for more details.
```

> 1. Identify the interface names:

Reviewer: ```suggestion
1. Identify the interface names with the following command:
```

    ```
    ip a
    ```

> 1. Edit `/etc/sysconfig/network/ifcfg-mgmt-bo` to update the bonding slaves and bonding mode:

Reviewer: ```suggestion
1. Edit `/etc/sysconfig/network/ifcfg-mgmt-bo` and update the lines associated with bonding slaves and bonding mode:
```

    ```
    BONDING_SLAVE_0='ens5'
    BONDING_SLAVE_1='ens6'
    BONDING_MODULE_OPTS='miimon=100 mode=balance-tlb '
    ```

> 1. Restart the network:

Reviewer: ```suggestion
1. Restart the network with the `wicked ifreload` command:
```

    ```
    wicked ifreload mgmt-bo
    ```

    :::caution

>     A misconfiguration might disrupt the ssh session.

Reviewer: ```suggestion
    A mistake in the configuration may disrupt the SSH session.
```

    :::

### Configuration persistence

> 1. Backup the elemental cloud-init file: `/oem/90_custom.yaml`:

Reviewer: ```suggestion
1. Backup the elemental cloud-init file `/oem/90_custom.yaml` as follows:
```
